packages/storage/src/indexed-db/index.ts

```diff
@@ -580,11 +585,31 @@ export class IndexedDb implements IndexedDbInterface {
     });
   }
 
-  async getPricesForAsset(assetId: AssetId): Promise<EstimatedPrice[]> {
-    const base64AssetId = uint8ArrayToBase64(assetId.inner);
+  /**
+   * Uses priceRelevanceThreshold to return only actual prices
+   * If more than priceRelevanceThreshold blocks have passed since the price was saved, such price is not returned
+   * priceRelevanceThreshold depends on the type of assets, for example, for delegation tokens the relevance lasts longer
+   */
+  async getPricesForAsset(
+    assetMetadata: Metadata,
+    latestBlockHeight: bigint,
+  ): Promise<EstimatedPrice[]> {
+    const base64AssetId = uint8ArrayToBase64(getAssetId(assetMetadata).inner);
     const results = await this.db.getAllFromIndex('PRICES', 'pricedAsset', base64AssetId);
 
-    return results.map(price => EstimatedPrice.fromJson(price));
+    const priceRelevanceThreshold = this.determinePriceRelevanceThresholdForAsset(assetMetadata);
+    const minHeight = latestBlockHeight - BigInt(priceRelevanceThreshold);
+
+    return results
+      .map(price => EstimatedPrice.fromJson(price))
+      .filter(price => price.asOfHeight >= minHeight);
+  }
```

Comment:
I decided to do this in-memory filtering because we won't have a lot of price records for each asset. And using a cursor or double index would make the code and its support much more complicated